From edcf6257ae2c88cc49010262f77f26494f0a6ddc Mon Sep 17 00:00:00 2001
From: Sergei Miroshnichenko <s.miroshnichenko@yadro.com>
Date: Tue, 22 Sep 2020 12:33:25 +0300
Subject: [PATCH 37/57] powerpc/powernv/pci: Don't print PHB registers dump
 when EEH is disabled

When EEH is replaced with standard PCIe facilities, errors on the bus still
trigger a PHB registers dump printing from skiboot, which slows down the
system unacceptably. So disable it when not needed.

Signed-off-by: Sergei Miroshnichenko <s.miroshnichenko@yadro.com>
---
 arch/powerpc/platforms/powernv/eeh-powernv.c |  3 +++
 arch/powerpc/platforms/powernv/pci.c         | 12 +++++++-----
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/arch/powerpc/platforms/powernv/eeh-powernv.c b/arch/powerpc/platforms/powernv/eeh-powernv.c
index 111111111..111111111 100644
--- a/arch/powerpc/platforms/powernv/eeh-powernv.c
+++ b/arch/powerpc/platforms/powernv/eeh-powernv.c
@@ -517,6 +517,9 @@ static void pnv_eeh_get_phb_diag(struct eeh_pe *pe)
 	struct pnv_phb *phb = pe->phb->private_data;
 	s64 rc;
 
+	if (!eeh_enabled())
+		return;
+
 	rc = opal_pci_get_phb_diag_data2(phb->opal_id, pe->data,
 					 phb->diag_data_size);
 	if (rc != OPAL_SUCCESS)
diff --git a/arch/powerpc/platforms/powernv/pci.c b/arch/powerpc/platforms/powernv/pci.c
index 111111111..111111111 100644
--- a/arch/powerpc/platforms/powernv/pci.c
+++ b/arch/powerpc/platforms/powernv/pci.c
@@ -488,14 +488,16 @@ void pnv_pci_dump_phb_diag_data(struct pci_controller *hose,
 static void pnv_pci_handle_eeh_config(struct pnv_phb *phb, u32 pe_no)
 {
 	unsigned long flags, rc;
-	int has_diag, ret = 0;
+	int has_diag = 0, ret = 0;
 
 	spin_lock_irqsave(&phb->lock, flags);
 
-	/* Fetch PHB diag-data */
-	rc = opal_pci_get_phb_diag_data2(phb->opal_id, phb->diag_data,
-					 phb->diag_data_size);
-	has_diag = (rc == OPAL_SUCCESS);
+	if (eeh_enabled()) {
+		/* Fetch PHB diag-data */
+		rc = opal_pci_get_phb_diag_data2(phb->opal_id, phb->diag_data,
+						 phb->diag_data_size);
+		has_diag = (rc == OPAL_SUCCESS);
+	}
 
 	/* If PHB supports compound PE, to handle it */
 	if (phb->unfreeze_pe) {

